genie-agent/src/main/java/com/netflix/genie/agent/execution/statemachine/actions/ShutdownAction.java

```diff
@@ -45,7 +58,25 @@
     @Override
     protected Events executeStateAction(final ExecutionContext executionContext) {
         log.info("Shutting down...");
-        //TODO implement this action
+
+        try {
+            log.info("Attempting to archive job folder...");
+
+            final JobSpecification jobSpecification = executionContext.getJobSpecification();
+            final String archiveLocation = jobSpecification.getArchiveLocation();
+            if (!StringUtils.isBlank(archiveLocation)) {
```

Comment:
Theres a `isNotBlank` method in `StringUtils` would help readability instead of using the `!` operator here